frontend/src/metabase/query_builder/components/SidebarHeader/SidebarHeader.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useMemo } from "react";
+import { t } from "ttag";
+import Icon from "metabase/components/Icon";
+import {
+  HeaderRoot,
+  HeaderIcon,
+  HeaderTitleContainer,
+  HeaderTitleContainerVariant,
+  CloseButton,
+} from "./SidebarHeader.styled";
+
+type Props = {
+  className?: string;
+  title?: string;
+  icon?: string;
+  onBack?: () => void;
+  onClose?: () => void;
+};
+
+function SidebarHeader({ className, title, icon, onBack, onClose }: Props) {
+  const hasDefaultBackButton = useMemo(() => !title && onBack, [title, onBack]);
+
+  const headerVariant = useMemo<HeaderTitleContainerVariant>(() => {
```

Comment:
Same with `useMemo` here. It might be useful just to extract this code to functions alongside this component